gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlanListDeserializer.java

```diff
@@ -107,6 +110,16 @@ public JobExecutionPlanListDeserializer(Map<URI, TopologySpec> topologySpecMap)
 
       JobExecutionPlan jobExecutionPlan = new JobExecutionPlan(jobSpec, specExecutor);
       jobExecutionPlan.setExecutionStatus(executionStatus);
+
+      try {
+        String jobExecutionFuture = serializedJobExecutionPlan.get(SerializationConstants.JOB_EXECUTION_FUTURE).getAsString();
+        Future future = specExecutor.getProducer().get().deserializeAddSpecResponse(jobExecutionFuture);
+        jobExecutionPlan.setJobFuture(Optional.fromNullable(future));
+
+      } catch (ExecutionException | InterruptedException e) {
```

Comment:
Maybe propagate the exception - instead of silently logging and returning?

Reply:
done